### PR TITLE
Use web.config environment variables to read if in Development

### PR DESCRIPTION
**Pull request by jkotalik**
Caught this during verification. By default, the ASPNETCORE_ENVIRONMENT is set in the web.config rather than on the iisexpress process, which I thought to be the case.

### CODE REVIEW

**Comment by anurse** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=205464

File: src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/ShimOptions.cpp [hunk 1]
```diff
@@ -43,11 +43,17 @@ ShimOptions::ShimOptions(const ConfigurationSource &configurationSource) :
...
-    // Reading environment variables can be added here, but it adds more code to the shim.
-    const auto detailedErrors = Environment::GetEnvironmentVariableValue(L"ASPNETCORE_DETAILEDERRORS").value_or(L"");
-    const auto aspnetCoreEnvironment = Environment::GetEnvironmentVariableValue(L"ASPNETCORE_ENVIRONMENT").value_or(L"");
-    const auto dotnetEnvironment = Environment::GetEnvironmentVariableValue(L"DOTNET_ENVIRONMENT").value_or(L"");
+    auto environmentVariables = section->GetMap(CS_ASPNETCORE_ENVIRONMENT_VARIABLES);
+
+    // Indexing into environment variable map will add a default entry if none is present
+    // This is okay here as we throw away the map shortly after.
+    // Process set environment variables are prioritized over web config variables.
+    const auto detailedErrors = Environment::GetEnvironmentVariableValue(L"ASPNETCORE_DETAILEDERRORS")
```
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/ShimOptions.cpp]: nit: store these names somewhere

**Review by BrennanConroy — APPROVED** [review]

**Comment by jkotalik** [comment]
@aspnet/build it seems that my PR lost the PR check for helix. This a known issue?

**Comment by jkotalik** [comment]
Pr is green here: https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190530.15

**Pull request merged**